[docs] Add documents for KoP's Schema Registry

BewareMyPower opened the pull request:
Fixes https://github.com/streamnative/kop/issues/1673

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [x] `doc` 
  
  (If this PR contains doc changes)

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+> **Note**
+>
+> See [here](./configuration.md) for other configurations, like how to change the port it listens on.
```
```suggestion
> For other configurations, such as how to change the port where the Schema Registry service listens on, see the [configuration guide](./configuration.md).
```

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+## Example: Use KoP Schema Registry with Confluent's Avro serializer
+
+This section adds an example to show how to use Confluent's Avro serializer on KoP.
```
```suggestion
This section provides an example about how to use Confluent's Avro serializer on KoP.
```

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+This section adds an example to show how to use Confluent's Avro serializer on KoP.
+
+### Enable Schema Registry in KoP standalone
```
```suggestion
### Enable the Schema Registry service on KoP
```

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+### Enable Schema Registry in KoP standalone
+
+Follow the **Set configuration for KoP** section in [here](./kop.md), start the KoP standalone with the following configurations:
```
```suggestion
Start KoP with the following properties in standalone mode. For details, see [Set configuration for KoP](./kop.md#set-configuration-for-kop).
```

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+### Define the Avro schema
+
+
+Create an Avro schema file named `User.asvc` under the `src/main/avro` directory:
```
Does this sentence and the sentence below has a sequential order? If yes, i would like to suggesting using an steplist format. Otherwise, ignore my comment.

BewareMyPower replied:
No.

Huanli-Meng commented on docs/schema.md:
````diff
@@ -0,0 +1,186 @@
+}
+```
+
+Add the `avro-maven-plugin` in order to generate the Java class from the Avro Schema file above.
````
```suggestion
Add the `avro-maven-plugin` plugin to generate the Java class from the Avro schema file above.
```

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+### Basic end-to-end example with Confluent's Avro serializer
+
+Add the following dependencies to use Kafka client 3.3.1 with Confluent's Avro serializer 7.3.1:
```
same comment as in the above section.

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+### Query the created schema
+
+Query the schema whose id is 1 (i.e. the 1st schema):
```
```suggestion
Query the schema whose ID is 1 (i.e. the 1st schema):
```

Huanli-Meng commented on docs/schema.md:
````diff
@@ -0,0 +1,186 @@
+}
+```
+
+See [here](https://docs.confluent.io/platform/current/schema-registry/schema_registry_tutorial.html#terminology-review) for a quick start of Confluent's Schema concepts.
````
```suggestion
For a quick start of Confluent's Schema concepts, See [Confluent documentation](https://docs.confluent.io/platform/current/schema-registry/schema_registry_tutorial.html#terminology-review) .
```
If this sentence belongs to the query created schema section, it is OK to put the sentence here. If it is just a reference that you want users to get more info, it would be great to use a format, such as:
## References
For a quick start of Confluent's Schema concepts, See [Confluent documentation](https://docs.confluent.io/platform/current/schema-registry/schema_registry_tutorial.html#terminology-review) .

Huanli-Meng commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+To manage the schemas, see the [Confluent's Schema REST API](https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility).
+
+## Example: Use KoP Schema Registry with Confluent's Avro serializer
```
It seems that your example contains the following steps:

1. Enable the Schema Registry service on KoP
2. Define the Avro schema
3. Basic end-to-end example with Confluent's Avro serializer
4. Query the created schema

Among these 4 steps, the 3rd one is not a task title (I mean using a verb + noun format). It would be better to update the title for step 3. Therefore, these 4 sections keep the same format.

Huanli-Meng reviewed (commented):
Leave some comments. PTAL. Feel free to let me know if there is any question.

BewareMyPower commented on docs/configuration.md:
```diff
@@ -180,3 +180,11 @@ This section lists configurations about the authentication.
+| Name | Description | Default |
+| ---- | ----------- | ------- |
+| kopSchemaRegistryEnable | Whether to start the Schema Registry service | false |
+| kopSchemaRegistryPort | The Schema Registry port | 8001 |
```
The [port](https://en.wikipedia.org/wiki/Port_(computer_networking)) is which the Schema Registry listens on, I think we should not use an ID to represent it.

BewareMyPower commented on docs/schema.md:
```diff
@@ -0,0 +1,186 @@
+
+KoP has implemented the Schema Registry to support Kafka clients with [Confluent's Schema serializers and deserializers](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#supported-formats).
+
+To enable the schema registry, you should add the following configuration:
```
Just like other KoP configurations. If you start a broker with KoP, add it to `broker.conf`. If you start a standalone with KoP, add it to `standalone.conf`. So I think it would be redundant to mention which `.conf` file should it be added to.

BewareMyPower commented:
@Huanli-Meng I have addressed the comments, PTAL again. I found the original API docs mix the terms of `Schema Registry` and `Schema Registry service`. I prefer the first one because Confluent's docs do not add the `service`.

Huanli-Meng reviewed (dismissed):
Overall LGTM and only two minor comments.

BewareMyPower commented:
@Huanli-Meng All comments are addressed now.

Huanli-Meng reviewed (approved):
LGTM. Thanks
